org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/TaskNodeViewerSorter.java

```diff
@@ -78,25 +82,30 @@ public int compare(Viewer viewer, Object leftNode, Object rightNode) {
      * @return the new sorter instance
      */
     public static TaskNodeViewerSorter createFor(TaskViewState state) {
-        Ordering<ProjectNode> projectOrdering = createProjectNodeOrdering();
+        Ordering<ProjectNode> projectOrdering = createProjectNodeOrdering(state.isProjectHierarchyFlattened());
         Ordering<TaskNode> taskOrdering = createTaskNodeOrdering(state.isSortByType(), state.isSortByVisibility());
         return new TaskNodeViewerSorter(projectOrdering, taskOrdering);
     }
 
-    private static Ordering<ProjectNode> createProjectNodeOrdering() {
+    private static Ordering<ProjectNode> createProjectNodeOrdering(final boolean isHierarchyFlattened) {
         return new Ordering<ProjectNode>() {
 
             @Override
             public int compare(ProjectNode left, ProjectNode right) {
-                EclipseProject leftRoot = HierarchicalElementUtils.getRoot(left.getEclipseProject());
-                EclipseProject rightRoot = HierarchicalElementUtils.getRoot(right.getEclipseProject());
-                if (leftRoot == rightRoot) {
```

Comment:
The root project which `HierarchicalElementUtils.getRoot` returned isn't the same instance, but actual is the same gradle project.

Reply:
That's correct, thanks for the explanation.